src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -114,17 +117,29 @@
             """)
     List<ProgrammingExercise> findAllWithStudentParticipationByRecentEndDate(@Param("endDate1") ZonedDateTime endDate1, @Param("endDate2") ZonedDateTime endDate2);
 
-    @Query("SELECT DISTINCT pe FROM ProgrammingExercise pe LEFT JOIN FETCH pe.studentParticipations pep LEFT JOIN FETCH pep.submissions")
-    List<ProgrammingExercise> findAllWithEagerParticipationsAndSubmissions();
+    @Query("""
+            SELECT DISTINCT pe FROM ProgrammingExercise pe
+            LEFT JOIN FETCH pe.studentParticipations pep
+            LEFT JOIN FETCH pep.submissions s
+            WHERE (s.type <> 'ILLEGAL' OR s.type IS NULL)
```

Comment:
Do we need the `OR s.type IS NULL` part? As far as I understand if `s.type` is null, then it therefore is not ILLEGAL and the condition already evaluates to true. Am I missing something here?

Reply:
Apparently we do need it but @derLalla knows more about why we need it.

Reply:
Nope, it is different in SQL Queries. Null cannot be compared to true or false. If s.type is null, `s.type <> 'ILLEGAL' evaluates to UNKNOWN and breaks the query. The found element is therefore not added to the SQL result.

Also we need that because of the testcases. There are several testcases where a submission is created to test e.g. just the results attached to it, but the type was never defined. This will never happen in the production environment but it is possible in the testcases, because it is an optional value.
This is not limited to the SubmissionType, but this is the case for a lot of entities.

Reply:
Ah ok, I didn't konw that but it makes sense :+1: Good to know :D